rpc/server.go

```diff
@@ -179,8 +181,14 @@ func (s *Server) serveRequest(codec ServerCodec, singleShot bool, options CodecO
 	for atomic.LoadInt32(&s.run) == 1 {
 		reqs, batch, err := s.readRequest(codec)
 		if err != nil {
-			log.Debug(fmt.Sprintf("read error %v\n", err))
-			codec.Write(codec.CreateErrorResponse(nil, err))
+			// If a parsing error occurred, send an error and terminate
+			if err.Error() != "EOF" {
```

Comment:
Special handling will be needed anyway, otherwise we'll send back an error json when the stream terminates, which arguably is not an error condition.